rust/src/services/messages/message_parser.rs

```diff
@@ -58,16 +59,17 @@ impl MessageParserService {
     }
 }
 
-/// Request type for the [`MessageParserService`].
-///
-/// It contains the encrypted message.
 #[derive(From, Debug)]
-pub struct MessageParserRequest(Vec<u8>);
-
-/// Response type for the [`MessageParserService`].
-///
-/// It contains the parsed message.
-pub type MessageParserResponse = Result<MessageOwned, MessageParserError>;
+pub struct RawMessage<T: AsRef<[u8]>>(T);
+
+impl<T> Traceable for RawMessage<T>
+where
+    T: AsRef<[u8]>,
+{
+    fn make_span(&self) -> Span {
+        error_span!("raw_message", payload_len = self.0.as_ref().len())
```

Comment:
What is the reason for using the `error_span!` instead of using `info_span!`?

Reply:
If the log level is `trace`, `warn` or `info`, there's no difference. But if the log level is `warn` or `error`, the span created by `info_span!` would be empty. By using `error_span!` we're basically saying: "whatever the log level is, make this a non-empty span".

Reply:
Ahh I see
Thank you!